superset/datasets/commands/update.py

```diff
@@ -98,8 +98,9 @@ def validate(self) -> None:
             exceptions.append(DatabaseChangeValidationError())
         # Validate/Populate owner
         try:
-            owners = self.populate_owners(self._actor, owner_ids)
-            self._properties["owners"] = owners
+            if owner_ids:
+                owners = self.populate_owners(self._actor, owner_ids)
+                self._properties["owners"] = owners
```

Comment:
Hugh, this will make the dataset update command behave differently than all other commands.

Why are we not passing the list of current owners when doing the `PUT` request? That would solve the problem.